test/integration/backward_compatible/parallel/sql/DataTypeTest.js

```diff
@@ -814,4 +815,59 @@ describe('Data type test', function () {
             rows[i]['__key'].should.be.eq(expectedKeys[i]);
         }
     });
+    it('should be able to decode/serialize JSON', async function () {
+        await basicSetup(this);
+        const SqlColumnType = TestUtil.getSqlColumnType();
+
+        // JSON support is added in 5.1.
+        TestUtil.markClientVersionAtLeast(this, '5.1');
+        TestUtil.markServerVersionAtLeast(this, client, '5.1');
+
+        const createMappingQuery = `
+            CREATE MAPPING ${mapName} (
+                __key INT,
+                this JSON
+            )
+            TYPE IMAP
+            OPTIONS (
+                'keyFormat' = 'int',
+                'valueFormat' = 'json'
+            )
+        `;
+
+        await client.getSql().execute(createMappingQuery);
+
+        await client.getSql().execute(`INSERT INTO ${mapName} VALUES (2, CAST('{"age": 2}' AS JSON)),` +
+            '(3, ?)',
+        [new HazelcastJsonValue(JSON.stringify({age: 3}))]);
+
+        const result = await TestUtil.getSql(client).execute(
+            `SELECT * FROM ${mapName} WHERE CAST(JSON_VALUE(this, '$.age') AS DOUBLE) > ?
+             AND CAST(JSON_VALUE(this, '$.age') AS DOUBLE) < ? ORDER BY __key ASC`,
+            [1, 4]
+        );
+
+        const rowMetadata = await TestUtil.getRowMetadata(result);
+        rowMetadata.getColumn(rowMetadata.findColumn('this')).type.should.be.eq(SqlColumnType.JSON);
+
+        const rows = [];
+
+        for await (const row of result) {
+            rows.push(row);
+        }
+
+        const expectedKeys = [2, 3];
+        const expectedBaseValues = {
+            age: 2
+        };
+        rows.length.should.be.eq(expectedKeys.length);
+
+        for (let i = 0; i < rows.length; i++) {
+            const jsonValue = rows[i]['this'];
+            jsonValue.should.be.instanceof(HazelcastJsonValue);
+            const obj = JSON.parse(jsonValue.toString());
+            obj.age.should.be.eq(expectedBaseValues.age + i);
+            rows[i]['__key'].should.be.eq(expectedKeys[i]);
+        }
+    });
```

Comment:
Can we also add a test with JS objects?

Reply:
done. the return type does not change, I added passing js object as parameter.